src/region.jl

```diff
@@ -0,0 +1,50 @@
+in_region(x, Y::Interval) = in_interval(x, Y)
+in_region(x::AbstractVector, Y::AbstractVector{<:Interval}) = all(in_interval.(x, Y))
+
+function intersect_region(X::Interval, Y::Interval)
+    intersection = intersect_interval(bareinterval(X), bareinterval(Y))
+    dec = min(decoration(X), decoration(Y))
+    guarantee = isguaranteed(X) && isguaranteed(Y)
+    return IntervalArithmetic._unsafe_interval(intersection, dec, guarantee)
+end
```

Comment:
I think this may be simplified to

```Julia
function intersect_region(X::Interval, Y::Interval)
    dec = min(decoration(X), decoration(Y))
    return IntervalArithmetic.setdecoration(intersect_interval(X, Y), dec)
end
```

Reply:
It can, indeed.